libglusterfs/src/statedump.c

```diff
@@ -300,28 +300,51 @@ gf_proc_dump_xlator_mem_info_only_in_use(xlator_t *xl)
     return;
 }
 
-/* Currently this dumps only mallinfo. More can be built on here */
+/* Currently this dumps only mallinfo. More can be built on here. */
+
+#if defined(HAVE_MALLINFO2)
+
+#define GF_MALLINFO()                                                          \
+    struct mallinfo2 info;                                                     \
+    memset(&info, 0, sizeof(info));                                            \
+    info = mallinfo2()
```

Comment:
This doesn't seem good to me. You should create a type that maps to `struct mallinfo2` or `struct mallinfo`, and a function to fill it.